src/Core/src/PropertyMapper.cs

```diff
@@ -93,35 +126,55 @@ public IPropertyMapper[]? Chained
 			}
 		}
 
-		private HashSet<string> PopulateKeys()
-		{
-			var keys = new HashSet<string>(StringComparer.Ordinal);
-			foreach (var key in GetKeys())
-			{
-				keys.Add(key);
-			}
-			return keys;
-		}
-
+		// TODO: Make private in .NET10 with a new name: ClearMergedMappers
 		protected virtual void ClearKeyCache()
 		{
-			_updateKeys = null;
+			_mergedMappers = null;
+			_mergedKeys = null;
 		}
 
-		public virtual IReadOnlyCollection<string> UpdateKeys => _updateKeys ??= PopulateKeys();
+		// TODO: Remove in .NET10
+		public virtual IReadOnlyCollection<string> UpdateKeys => MergedKeys;
 
 		public virtual IEnumerable<string> GetKeys()
 		{
-			foreach (var key in _mapper.Keys)
-				yield return key;
-
+			// We want to retain the initial order of the keys to avoid race conditions
+			// when a property mapping is overridden by a new instance of property mapper.
+			// As an example, the container view mapper should always run first.
+			// Siblings mapper should not have keys intersection.
 			if (Chained is not null)
 			{
-				foreach (var chain in Chained)
-					foreach (var key in chain.GetKeys())
+				for (int i = Chained.Length - 1; i >= 0; i--)
```

Comment:
Here, can flatten the nested loops:
```
foreach (var key in Chained.Reverse().SelectMany(chain => chain.GetKeys()))
{
    yield return key;
}
```

But need to pass benchmarks to verify if could improve the performance or not.

Reply:
Nothing beats a pure `for (int i` loop :)

LINQ version results:

| Method                    | Mean     | Error    | StdDev   | Gen0      | Allocated  |
|-------------------------- |---------:|---------:|---------:|----------:|-----------:|
| BenchmarkUpdateProperties | 64.74 ms | 0.947 ms | 0.840 ms | 3625.0000 | 31204126 B |
| BenchmarkUpdateProperty   | 12.64 ms | 0.065 ms | 0.058 ms |         - |       12 B |